src/plugins/condition/components/Criterion.vue

```diff
@@ -0,0 +1,145 @@
+                   v-model="comparisonInputValue"
+                   class="t-condition-name-input"
+                   type="text"
+                   @change="persist"
+            >
+        </span>
+    </span>
+</li>
+
+</template>
+
+<script>
+import { OPERATIONS } from '../utils/operations';
+
+export default {
+    inject: ['openmct'],
+    props: {
+        criterion: {
+            type: Object,
+            required: true
+        },
+        telemetry: {
+            type: Array,
+            required: true,
+            default: () => []
+        },
+        index: {
+            type: Number,
+            required: true
+        },
+        condition: {
+            type: Object,
+            required: true
+        },
+        trigger: {
+            type: String,
+            required: true
+        }
+    },
+    data() {
+        return {
+            telemetryMetadata: {},
+            operations: OPERATIONS,
+            selectedTelemetryObject: '',
+            selectedFieldOption: {},
+            selectedOperationOption: '',
+            operationValue: '',
+            comparisonInputValue: '',
+            isInputOperation: false,
+            rowLabel: ''
+        }
+    },
+    computed: {
+        setRowLabel: function () {
+            let operator = this.trigger === 'all' ? 'and ': 'or ';
+
+            return (this.index !== 0 ? operator : '') + 'when';
+        }
+    },
+    mounted() {
+        this.initialize();
+    },
+    updated() {
+        this.persist();
+    },
+    methods: {
+        initialize() {
+            this.selectedTelemetryObject = this.criterion.telemetry;
+            this.selectedFieldOption = this.criterion.metadata;
+            this.selectedOperationOption = this.criterion.operation;
+            this.comparisonInputValue = this.criterion.input;
+            this.updateOperationInput();
+            this.updateFieldOptions();
+
+        },
+        updateFieldOptions() {
+            if (this.selectedTelemetryObject) {
+                this.openmct.objects.get(this.selectedTelemetryObject).then((telemetryObject) => {
+                    this.telemetryMetadata = this.openmct.telemetry.getMetadata(telemetryObject).values();
+                });
+            }
+        },
+        updateOperationInput() {
+            if (this.selectedOperationOption &&
+                (this.selectedOperationOption !== 'isUndefined' &&
+                 this.selectedOperationOption !== 'isDefined')) {
+                this.isInputOperation = true;
+            } else {
+                this.isInputOperation = false;
+                this.comparisonInputValue = '';
+            }
+        },
+        persist() {
+            this.criterion.telemetry = this.selectedTelemetryObject;
+            this.criterion.metadata = this.selectedFieldOption;
+            this.criterion.operation = this.selectedOperationOption;
+            this.criterion.input = this.comparisonInputValue;
```

Comment:
See comment about using this.criterion directly in the markup. Then you wouldn't need to do this assigning back and forth.

Reply:
done